src/controllers/theatre.js

```diff
@@ -0,0 +1,83 @@
+const Theatre = require("../models/theatres");
+const MovieTheatreMapping = require("../models/movieTheatreMappings");
+const { setCache, getCache } = require("../configs/cacheConfig");
+
+const getAllTheatres = async (req, res) => {
+  try {
+    const { city } = req.query;
+    let theatres = [];
+    if (city) {
+      let cachedData = await getCache(`theater_${city}`);
```

Comment:
should this data be cached ? I don't think the query to find theatre in a city is expensive.